sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/CheckpointFileManager.scala

```diff
@@ -277,7 +277,6 @@ class FileSystemBasedCheckpointFileManager(path: Path, hadoopConf: Configuration
       fs.delete(path, true)
     } catch {
       case e: FileNotFoundException =>
-        logInfo(s"Failed to delete $path as it does not exist")
```

Comment:
Just to clarify, this log was deleted per https://github.com/apache/spark/pull/31495#discussion_r571870795

Reply:
@xuanyuanking @HeartSaVioR not sure I get the full point here. We've removed the log which notified the user that the file was not able to remove and now we have a new log like this:
`logTrace(s"Removed metadata log file: $path")`
The plan is that if the user sees this log then it was successful and when nothing is in the log then it has failed?
Can you enlighten me please?

Reply:
I'm quoting my comment:

> FileSystemBasedCheckpointFileManager.delete logs INFO message when the file doesn't exist. FileContextBasedCheckpointFileManager.delete doesn't log any message though.
>
> Probably OK to remove INFO message on FileSystemBasedCheckpointFileManager.delete as FileSystemBasedCheckpointFileManager is used as a fail-back of FileContextBasedCheckpointFileManager (especially Spark logs warn message when failing back) and FileContextBasedCheckpointFileManager covers the major usage.

We decided to "blindly" call `delete()` without checking existence in prior, as file system will do it instead and we can save one op call. Leaving this log would provide false-alarm, as it's "expected" even if there's no file, so we have to delete the log line.

Reply:
In the meantime I was digging the code and now I see why the if was needed. Calling `delete` blindly is really a speedup.
False-alarm is the key to my question, thanks!

Reply:
> Calling delete blindly is really a speedup.

mmm. Those HEAD requests add up, don't they?

FWIW, Hadoop-3.3.1 adds a lot of IOStatistics collection on s3a and abfs operations, for which there's an interface implemented by, amongst other things: input streams, output streams, filesystems, remote iterators. But toString() logs this stuff too. So if you log an FS.toString() at debug you will see these. We are doing more than just counting #of calls, we are collecting min/max/mean performance. LIST is the killer -especially if you have versioning turned on on S3 and then do lots of overwrite/delete of objects.

Reply:
You're right, HEAD requests add up and I've formed my statement roughly.
According to my measurements the nominal performance better even with blind `delete`.
To tell the truth not measured independently how much caching brings and how much blind `delete` wastes.
If you have a good idea to make it better it's more than welcome.

Reply:
I've already experimented and noticed both will have no noticeable difference.
https://github.com/apache/spark/pull/31495#discussion_r572473687

But that's completely based on implementation details; implementations can leverage the benefit of doing less operations if external storage could simply do what the operation associated with delete promises the same contract delete() in Hadoop filesystem promises.